lunavl/sdk/estimators/face_estimators/head_pose.py

```diff
@@ -156,20 +156,43 @@ def estimate(self, landmarks68: Landmarks68) -> HeadPose:  # type: ignore
         return self.estimateBy68Landmarks(landmarks68)
 
     @CoreExceptionWrap(LunaVLError.EstimationHeadPoseError)
-    def estimateByBoundingBox(self, detection: BoundingBox, imageWithDetection: VLImage) -> HeadPose:
+    def estimateByBoundingBox(self, imageWithFaceDetection: ImageWithFaceDetection) -> HeadPose:
         """
         Estimate head pose by detection.
 
         Args:
-            detection: detection bounding box
-            imageWithDetection: image with the detection.
+            imageWithFaceDetection: image with face detection
         Returns:
             estimate head pose
         Raises:
             LunaSDKException: if estimation is failed
         """
-        error, headPoseEstimation = self._coreEstimator.estimate(imageWithDetection.coreImage, detection.coreEstimation)
+        error, headPoseEstimation = self._coreEstimator.estimate(
+            imageWithFaceDetection.image.coreImage, imageWithFaceDetection.boundingBox.coreEstimation
+        )
 
         if error.isError:
             raise LunaSDKException(LunaVLError.fromSDKError(error))
         return HeadPose(headPoseEstimation)
+
+    @CoreExceptionWrap(LunaVLError.EstimationHeadPoseError)
+    def estimateByBoundingBoxBatch(self, imageWithFaceDetectionList: List[ImageWithFaceDetection]) -> List[HeadPose]:
```

Comment:
estimateBatch(data: Union[ List[ImageWithFaceDetection],  List[FaceDetection]])

Reply:
ok

Reply:
rename function to `estimateBatch`

Reply:
done